tests/test_extra_files.py

```diff
@@ -10,38 +12,26 @@ def test_simple():
             'foo': 'abc',
         }
     }
-    assert dict(get_extra_files(config)) == {'foo': b'abc'}
+    assert list(get_extra_files(config)) == [('foo', 'content', b'abc')]
 
 def test_content():
     config = {
         'extra_files': {
-            'str': 'ábč',
-            'bytes': b'def',
-            'base64': {'base64': 'Z2hp'},
-            'copied': {
+            'str.txt': 'ábč',
+            'bytes.dat': b'def',
+            'base64.dat': {'base64': 'Z2hp'},
+            'copied.png': {
                 'copy_from': FIXTURES_PATH / 'app_with_extra_files/smile2.png'
             },
```

Comment:
Take as example to explain the differ between url_part in application and path to the file, which should be used as static file.